### PR TITLE
Add a diagnostic script to assist in the debugging of issues with appsmith implementation

### LINKED ISSUE #35639: [Feature]: add diagnostic script to gather troubleshooting info
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Create a script to gather maximum diagnostic info from customer implementation

### Why should this be worked on?

Assist in the troubleshooting of appsmith instances

### PR DESCRIPTION
**Pull request by zbuildz**
## Description
A script to gather logs and system info into a tarball which can be extracted and analyzed by appsmith engineers.

Fixes #35639 #32971

## Automation

/ok-to-test tags=""

### :mag: Cypress test results

> [!WARNING]
> Tests have not run on the HEAD 4f8e27cd8bcef9233786f268f95e0b3e981c13d6 yet
> <hr>Tue, 27 Aug 2024 14:01:33 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a new diagnostic tool to gather system and application metrics for improved troubleshooting and performance monitoring in the Appsmith environment.
	- The tool collects logs, configuration files, JVM properties, and system-level information, compressing the data for easier transport.

- **Chores**
	- Added a script for efficient collection and organization of diagnostic data, enhancing operational capabilities.

### CODE REVIEW

**Review by pratapaprasanna — CHANGES_REQUESTED** [review]
Please add alerts for mongo, redis and if there is enough space ?
etc. 
Also, I know we are taking backup of docker.env . Lets please remove mongo creds from the env file and take a copy

File: deploy/docker/fs/opt/appsmith/diagnostics.sh [hunk 1]
```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env bash
+
+modified_within_last_x_minutes=180
+tail_lines=10000
+hostname=$(hostname)
+timestamp=$(date +%F_%H.%M.%S-%Z)
+tmpdir=tmpdir="$(mktemp -d)/"$hostname"/"$timestamp""
```
Comment (pratapaprasanna) [inline deploy/docker/fs/opt/appsmith/diagnostics.sh:7]: We dont have to create another tmp directory. you can use 
`$TMP` env variable

File: deploy/docker/fs/opt/appsmith/diagnostics.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+#
+
+# gather the container-info
+cp /opt/appsmith/info.json "$tmpdir/container-info.json"
+
+# gather the infra-details
+cp /tmp/appsmith/infra.json "$tmpdir/infra-info.json"
+
+# gather the healthcheck
+/opt/appsmith/healthcheck.sh > "$tmpdir/healthcheck.txt"
```
Comment (pratapaprasanna) [inline deploy/docker/fs/opt/appsmith/diagnostics.sh:48]: I think we are covering healthchecks here

**Review by pratapaprasanna — CHANGES_REQUESTED** [review]
please refrain from using tmp directory. How about using the tmp directory we already have in the container.
You can access the path with $TMP and access it.
```
root@appsmith-678869d698-m6d8x:/opt/appsmith# echo $TMP
/tmp/appsmith
```
and write required contents to that path. 

I am good with the rest of the things.

File: deploy/docker/fs/opt/appsmith/diagnostics.sh [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+# Log info
+#
+
+# gather the logs
+for i in `find /appsmith-stacks/logs/* -type f -mmin -"$modified_within_last_x_minutes"`; do
+    if [[ -e "$i" ]]; then
+        mkdir -p "$tmpdir/`dirname ${i:1}`"
+        tail -"$tail_lines" "$i" > "$tmpdir/${i:1}"
+    fi
+done
```
[the remark this answers is not in the document]
Comment (zbuildz) [reply deploy/docker/fs/opt/appsmith/diagnostics.sh:35]: modified per coderabbit suggestion

**Comment by zbuildz** [comment]
updated changes per coderabbit and @pratapaprasanna. mongodb, redis and postgres checks will be added in future iterations

**Review by pratapaprasanna — APPROVED** [review]

**Pull request merged**